temporal-sdk/src/test/java/io/temporal/worker/ActivityPollerDisabledTest.java

```diff
@@ -0,0 +1,86 @@
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+package io.temporal.worker;
+
+import static java.util.stream.Collectors.groupingBy;
+import static org.junit.Assert.*;
+
+import io.temporal.testing.TestEnvironmentOptions;
+import io.temporal.testing.TestWorkflowEnvironment;
+import io.temporal.workflow.WorkflowInterface;
+import io.temporal.workflow.WorkflowMethod;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.junit.Test;
+
+public class ActivityPollerDisabledTest {
+
+  @WorkflowInterface
+  public interface Workflow {
+    @WorkflowMethod
+    void bar();
+  }
+
+  public static class WorkflowImpl implements Workflow {
+
+    @Override
+    public void bar() {}
+  }
+
+  @Test
+  public void testActivityPollerDisabled() throws InterruptedException {
+    String activityPollerThreadNamePrefix = "Activity Poller task";
+    String workflowPollerThreadNamePrefix = "Workflow Poller task";
+    String workflowHostLocalPollerThreadNamePrefix = "Host Local Workflow ";
+    int hostLocalThreadCount = 22;
+    int workflowPollCount = 11;
+    int activityPollCount = 18;
+
+    TestEnvironmentOptions options =
+        TestEnvironmentOptions.newBuilder()
+            .setWorkerFactoryOptions(
+                WorkerFactoryOptions.newBuilder()
+                    .setWorkflowHostLocalPollThreadCount(hostLocalThreadCount)
+                    .build())
+            .build();
+    TestWorkflowEnvironment env = TestWorkflowEnvironment.newInstance(options);
+    Worker worker =
+        env.newWorker(
+            "tl1",
+            WorkerOptions.newBuilder()
+                .setWorkflowPollThreadCount(workflowPollCount)
+                .setActivityPollThreadCount(activityPollCount)
+                .setActivityPollerDisabled(true)
+                .build());
+    // Need to register something for workers to start
+    worker.registerWorkflowImplementationTypes(WorkflowImpl.class);
+    env.start();
+    Thread.sleep(1000);
+    Map<String, Long> threads =
+        Thread.getAllStackTraces().keySet().stream()
+            .map((t) -> t.getName().substring(0, Math.min(20, t.getName().length())))
+            .collect(groupingBy(Function.identity(), Collectors.counting()));
+    assertEquals(hostLocalThreadCount, (long) threads.get(workflowHostLocalPollerThreadNamePrefix));
+    assertEquals(workflowPollCount, (long) threads.get(workflowPollerThreadNamePrefix));
+    assertFalse(threads.containsKey(activityPollerThreadNamePrefix));
```

Comment:
This is a pretty neat trick. Do we have any business-logic-related thing we can verify here other than there are no threads and the worker is null?

Reply:
It's the last assertion line:
```
    assertNull(worker.activityWorker);
```

Reply:
Right - I meant is there something besides that we can assert, like something doesn't happen that otherwise would? Perhaps an integ test that verifies the worker doesn't attempt to perform any activities?

If you think it's low value don't bother. I was just curious if there's something we could do here besides "this is null"

Reply:
Another option is to schedule a local activity and after it completes schedule a normal activity and let it timeout with ScheduleToStart to show that no activity worker is present.

Reply:
Added two tests for local and normal activities.

Reply:
:tada: Thanks!